### DISCUSSION TITLE
PHP v8.3 zend stack size

### DISCUSSION THREAD

**Discussion by pag1903** [libvips/php-vips#234] [Q&A] [answered]
Hello,

I've just tried to use php-vips under PHP v8.3 but I've encounter an Error due to [new PHP settings](https://www.php.net/manual/es/migration83.other-changes.php#migration83.other-changes.ini) (see [PR](https://github.com/php/php-src/pull/9104)):

> zend.max_allowed_stack_size is a new INI directive to set the maximum allowed stack size. Possible values are 0 (detect the process or thread maximum stack size), -1 (no limit), or a positive number of bytes. The default is 0. When it is not possible to detect the process or thread maximum stack size, a known system default is used. Setting this value too high has the same effect as disabling the stack size limit. Fibers use fiber.stack_size as maximum allowed stack size. An [Error](https://www.php.net/manual/es/class.error.php) is thrown when the process call stack exceeds zend.max_allowed_stack_size-zend.reserved_stack_size bytes, to prevent stack-overflow-induced segmentation faults, with the goal of making debugging easier. The stack size increases during uncontrolled recursions involving internal functions or the magic methods [__toString()](https://www.php.net/manual/es/language.oop5.magic.php#object.tostring), [__clone()](https://www.php.net/manual/es/language.oop5.cloning.php#object.clone), [__sleep()](https://www.php.net/manual/es/language.oop5.magic.php#object.sleep), [__destruct()](https://www.php.net/manual/es/language.oop5.decon.php#object.destruct). This is not related to stack buffer overflows, and is not a security feature.

Here is the backtrace:
```
Uncaught Error: Maximum call stack size of 8339456 bytes (zend.max_allowed_stack_size - zend.reserved_stack_size) reached. Infinite recursion? in /projet/workspace/00-dc-storage-quality_ci_quality/vendor/jcupitt/vips/src/GObject.php:153
Stack trace:
#0 /projet/workspace/00-dc-storage-quality_ci_quality/vendor/jcupitt/vips/src/VipsOperation.php(308): Jcupitt\Vips\GObject::Jcupitt\Vips\{closure}()
#1 /projet/workspace/00-dc-storage-quality_ci_quality/vendor/jcupitt/vips/src/VipsOperation.php(308): FFI->vips_cache_operation_build()
#2 /projet/workspace/00-dc-storage-quality_ci_quality/vendor/jcupitt/vips/src/VipsOperation.php(371): Jcupitt\Vips\VipsOperation::callBase()
#3 /projet/workspace/00-dc-storage-quality_ci_quality/vendor/jcupitt/vips/src/Image.php(1104): Jcupitt\Vips\VipsOperation::call()
#4 /projet/workspace/00-dc-storage-quality_ci_quality/src/Utils/VipsUtils.php(70): Jcupitt\Vips\Image->writeToTarget()
```

So I'm wondering what are your recommandations for this issue: disable those new settings (-1) or try to wrap the operation in Fiber (I've never test this yet) ?

**Comment by jcupitt** [comment, marked as answer]
Hi @pag1903,

There are a couple of PRs in progress now for php 8.3 support, eg.:

https://github.com/libvips/php-vips/pull/231

I'd follow those for updates.

**Reply by pag1903** [reply]
Thank you for your reply.
